lib/Firtool/Firtool.cpp

```diff
@@ -100,8 +100,11 @@ LogicalResult firtool::populateCHIRRTLToLowFIRRTL(mlir::PassManager &pm,
 
   pm.nest<firrtl::CircuitOp>().addPass(firrtl::createDropConst());
 
-  if (opt.shouldDedup())
-    pm.nest<firrtl::CircuitOp>().addPass(firrtl::createDedup());
+  if (opt.shouldDedup()) {
+    firrtl::DedupOptions opts;
+    opts.dedupClasses = opt.shouldDedupClasses();
+    pm.nest<firrtl::CircuitOp>().addPass(firrtl::createDedup(opts));
```

Comment:
This could be `createDedup({opt.shouldDedupClasses()})`.  We handle this a few ways throughout, this way has benefit of making it super clear what's going on.  If we had field designators (c++20) we could keep that readability benefit.

Anyway, whatever you like better :+1:.